lib/fog/libvirt/models/compute/templates/server.xml.erb

```diff
@@ -32,13 +32,38 @@
 <% end -%>
   <clock offset='utc'/>
   <devices>
+<% args = {}
+   if File.file?('/etc/foreman/ceph.conf')
+     File.readlines('/etc/foreman/ceph.conf').each do |line|
+       pair = line.strip.split("=")
+       key = pair[0]
+       value = pair[1]
+       args[key] = value
+     end
+   end
+%>
 <% volumes.each do |vol| -%>
+  <% if File.file?('/etc/foreman/ceph.conf') && vol.pool_name.include?(args["libvirt_ceph_pool"]) %>
+    <disk type='network' device='disk'>
+      <driver name='qemu' type='<%= vol.format_type %>' cache='writeback' discard='unmap'/>
+      <source protocol='rbd' name='<%= vol.path %>'>
+        <% args["monitor"].split(",").each do |mon| %>
+        <host name='<%= mon %>' port='<%= args["port"] %>'/>
+        <% end %>
+      </source>
+      <auth username='<%= args["auth_username"] %>'>
+        <secret type='ceph' uuid='<%= args["auth_uuid"] %>'/>
```

Comment:
You could also fetch the secret by it's name/usage with `<secret type="ceph" usage="foo-secret-name-foo"/>`. This makes the config much easier to read and easier to deploy without having to deal with the secret UUID.

Reply:
This is a good idea - will give it a go (not sure if I manage this week, though). Thanks!

Reply:
@Bluewind I think the new syntax is not yet understood by libvirt as shipped with CentOS 7, which is still in wide use nowadays (that probably is also the reason why the Ceph docs use the UUID notation still on https://docs.ceph.com/docs/nautilus/rbd/libvirt/ ). 
At least I tried:
```
      <auth type='ceph' username='libvirt'>
        <secret usage='client.libvirt secret'/>
      </auth>
```
just now on CentOS 7, and it tells me:
```
Error: XML document failed to validate against schema: Unable to validate doc against /usr/share/libvirt/schemas/domain.rng
Extra element devices in interleave
Element domain failed to validate content
```

So I think for portability /compatibility reasons, we have to stay with UUID for now.

Reply:
I see. Could you maybe add settings for both options and use the name only when it is set to a non-empty string? In our setup (Ubuntu 18.04) the name already works just fine. Also, if you add the option now, people can switch whenever their environment supports it and they don't have to wait for another patch.

Reply:
Good idea! Since I can not test myself, can you confirm the syntax I posted is correct? 
I will then implement this hopefully next month (I am travelling to conferences a lot currently).

Reply:
I don't have a system to test this on either so I can just comment based on reading the code. The code in the PR looks alright to me. Note that I do not have experience with foreman code and I do not actually know the content/types of most of the variables (especially `vol.pool_name`). I'm sure the fog devs will also double-check that the code makes sense before merging.

Reply:
The general implementation currently in the PR I can test, and this works well. My question was about the syntax example shown above:
```
    <disk type='network' device='disk'>
      <driver name='qemu' type='raw' cache='writeback' discard='unmap'/>
      <auth type='ceph' username='libvirt'>
        <secret usage='client.libvirt secret'/>
      </auth>
      <source protocol='rbd' name='rbd/test-vm.XXXX-disk1'>
        ....
      </source>
      <target dev='sda' bus='scsi'/>
      <address type='drive' controller='0' bus='0' target='0' unit='0'/>
    </disk>
```
If that works in this way for you outside of Foreman, I can integrate it in this PR and test if the XML is generated as expected - I just don't have a machine on which I can test the XML to start a VM.

Reply:
I see. I've tested your config in our setup and noticed that `type='ceph'` must be part of the `<secret>` element. If it is set in `<auth>`, I get `unsupported configuration: unknown secret type '<null>'`. 

Apart from that, the config works fine. Thanks!

Reply:
@bluewind: To spell it out, you mean:
```
    <disk type='network' device='disk'>
      <driver name='qemu' type='raw' cache='writeback' discard='unmap'/>
      <auth username='libvirt'>
        <secret type='ceph' usage='client.libvirt secret'/>
      </auth>
      <source protocol='rbd' name='rbd/test-vm.XXXX-disk1'>
        ....
      </source>
      <target dev='sda' bus='scsi'/>
      <address type='drive' controller='0' bus='0' target='0' unit='0'/>
    </disk>
```
works fine for you? This gives the same error for me on CentOS 7 as I stated above, so if you tell me this works, I'll take it for granted and implement it ;-).

Reply:
Weird. Yes, that configuration looks like the one I tested.

I'm currently on vacation so I can't test this, but I think you can just set the `type` attribute with both elements and then it will probably work in either case. I can test this in around 1-2 weeks if you want.

Reply:
@bluewind I think setting `type` on both must fail due to XML schema validation. Since the version in CentOS 7 does not seem to support `usage` at all, I will then implement as follows:
- If the user has specified a UUID, use that (compatible also with older versions). 
- If not, and a `usage` string has been specified, use that (works only with more recent distros). 

In both cases, leaving the `type` attribute as part of the `secret` tag should work just fine.

Reply:
@Bluewind This is implemented in the commit I pushed just now, including documentation. I tested that the XMLs are generated fine, as expected.

Reply:
After a quick look, I'd say that this looks good. Thanks!